mainwindow.cc

```diff
@@ -4199,57 +4193,45 @@ void MainWindow::showDictionaryInfo( const QString & id )
       }
       else if( result == DictInfo::SHOW_HEADWORDS )
       {
-        showDictionaryHeadwords( owner, dictionaries[x].get() );
+        showDictionaryHeadwords( dictionaries[x].get() );
       }
 
       break;
     }
   }
 }
 
-void MainWindow::showDictionaryHeadwords( const QString & id )
+void MainWindow::showDictionaryHeadwords( Dictionary::Class * dict )
 {
-  QWidget * owner = 0;
 
-  if( sender()->objectName().compare( "EditDictionaries" ) == 0 )
-    owner = qobject_cast< QWidget * >( sender() );
-
-  if( owner == 0 )
-    owner = this;
+  QWidget * owner = qobject_cast< QWidget * >( sender() );
 
-  for( unsigned x = 0; x < dictionaries.size(); x++ )
-  {
-    if( dictionaries[ x ]->getId() == id.toUtf8().data() )
-    {
-      showDictionaryHeadwords( owner, dictionaries[ x ].get() );
-      break;
-    }
-  }
-}
+  // DictHeadwords internally check parent== mainwindow to know why it is requested.
+  // If the DictHeadwords is requested by Edit->Dictionaries->ShowHeadWords, (owner = "EditDictionaries")
+  // it will be a modal dialog. When click a word, that word will NOT be queried.
+  // In all other cases, just set owner = mainwindow(this),
 
-void MainWindow::showDictionaryHeadwords( QWidget * owner, Dictionary::Class * dict )
-{
-  if( owner && owner != this )
-  {
+  if ( owner->objectName() == "EditDictionaries" ) {
     DictHeadwords headwords( owner, cfg, dict );
     headwords.exec();
-    return;
   }
```

Comment:
`return` directly in the `if` block,
this can reduce the follow `nest else` block depth.